base/intfuncs.jl

```diff
@@ -200,6 +200,38 @@ end
 # However, we still need a fallback that calls the general ^:
 ^{p}(x, ::Type{Val{p}}) = x^p
 
+# for numbers, inline the power_by_squaring method for literal p.
+# (this also allows integer^-p to give a floating-point result
+#  in a type-stable fashion).
+@generated function ^{p}(x::Number, ::Type{Val{p}})
```

Comment:
It seems analogous to saying we should default to `sum_kbn` because `sum` is inaccurate.  Yes, `sum_kbn` is slightly more accurate, but only slightly in most cases, and it is vastly slower.